### ISSUE TITLE
offset queries return wrong number of hits

### ISSUE THREAD

**Issue by niemannd** [meilisearch/MeiliSearch#947]
**Describe the bug**
The number of hits returned for a query with limit, offset and filter does not return an expected number of hits. 
Lets say you run a query with limit 2000, offset 0 and a filter (dont know yet if that plays a role in this bug) and meilisearch returns 688 hits (as in items in the "hits" array). Running the same query with offset 10 should now return only 678 hits, offset 20 returns 668 and so on. Now i have the case, that at some point the queries return more results than expected. 

Together with #764 this makes pagination basically impossible without loading the full resultset at once. 

**To Reproduce**
* fresh meilisearch instance
* setup movies index + load it with the [movies.json dataset](https://github.com/meilisearch/MeiliSearch/blob/master/datasets/movies/movies.json)
* run query with different offsets:
```
{ 
  "q": "b",
  "limit": 2000,
  "offset": 0,
  "filters": "release_date > 1420070400 AND release_date <= 1451606399"
}
```

| offset | returned hits | expected hits |------------- | 
| ------------- | ------------- |  ------------- | ------------- | 
| 0     | 688 | 688 | 
| 10   | 678 | 678 | 
| 20   | 668 | 668 | 
| 30   | 658 | 658 | 
| 40   | 648 | 648 | 
| 50   | 638 | 638 | 
| 60   | 628 | 628 | 
| 70   | 680 | 618 | ⚠️ 
| 80   | 670 | 608 | ⚠️ 
| 90   | 660 | 598 | ⚠️ 
| 100 | 616 | 588 | ⚠️ 

**Additional context**
Tested with Docker Image: 
getmeili/meilisearch:v0.14.0
getmeili/meilisearch:v0.13.0

**Comment by LegendreM** [comment]
Hello @niemannd,

does your bug only occurs with `filters` ?

**Comment by niemannd** [comment]
i don't know if it _only_ occurs with filters, but i haven't found a way to reproduce it without one. 

Another Example with the same dataset as above: 

```
{ 
	"q": "bal",
	"limit": 2000,
	"offset": 10,
	"filters": "release_date != 0"
}
```
Without a filter it returns the expected hit count, with filter the error starts at offset 20.
I asume it has something to do with filters.

**Comment by HelmutLety** [comment]
I do have the same issue, found any workaround?

**Comment by niemannd** [comment]
@HelmutLety we simply pull a big chunk of results at once (limit 1000 at the moment) with offset 0 and paginate over the array via javascript. The data is small enough that it doesn't have that much of an impact. its not pretty but it works for us.

**Comment by HelmutLety** [comment]
thanks @niemannd !
I can confirm that it occurs only with filters for us

**Comment by HelmutLety** [comment]
Some update, we don't have this bug with version v0.13.0

**Comment by MarinPostma** [comment]
this should be fixed with #849 then.

**Comment by qdequele** [comment]
@MarinPostma What is the status of this issue?

**Comment by MarinPostma** [comment]
Closed per #849
